examples/todos-postgres/src/main.rs

```diff
@@ -0,0 +1,111 @@
+use sqlx::PgPool;
+use std::env;
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+struct Args {
+    #[structopt(subcommand)]
+    cmd: Option<Command>,
+}
+
+#[derive(StructOpt)]
+enum Command {
+    Add { description: String },
+    Done { id: i64 },
+}
+
+#[async_std::main]
+#[paw::main]
+async fn main(args: Args) -> anyhow::Result<()> {
+    let mut pool = PgPool::new(&env::var("DATABASE_URL")?).await?;
+
+    match args.cmd {
+        Some(Command::Add { description }) => {
+            println!("Adding new todo with description '{}'", &description);
+            println!(
+                "Added new todo with id {}",
+                add_todo(&pool, &description).await.unwrap()
+            );
+        }
+        Some(Command::Done { id }) => {
+            println!("Marking todo {} as done", id);
+            if complete_todo(&pool, id).await.unwrap() {
+                println!("Todo {} is marked as done", id);
+            } else {
+                println!("Invalid id {}", id);
+            }
+        }
+        None => {
+            println!("Printing list of all todos");
+            list_todos(&mut pool).await.unwrap();
+        }
+    }
+
+    pool.close().await;
+
+    Ok(())
+}
+
+async fn add_todo(pool: &PgPool, description: &str) -> Result<i64, sqlx::error::Error> {
+    let mut tx = pool.begin().await?;
+
+    let rec = sqlx::query!(
+        r#"
+INSERT INTO todos ( description )
+VALUES ( $1 )
+RETURNING id
+        "#,
+        description.to_string()
+    )
+    .fetch_one(&mut tx)
+    .await?;
+
+    tx.commit().await?;
+
+    Ok(rec.id)
+}
+
+async fn complete_todo(pool: &PgPool, id: i64) -> Result<bool, sqlx::error::Error> {
+    let mut tx = pool.begin().await?;
+
+    let rows_affected = sqlx::query!(
+        r#"
```

Comment:
Note that `r#" ... #` is only needed if you want to use `"` or `\` in the SQL query. Normal strings can have line breaks.

Reply:
Ah, I didn't know that, I was just copying the other example 😅

Reply:
That just really clicked for me recently. It's one of those things were I have a fundamental belief that normal strings cannot have line breaks and I never even considered looking to see if Rust could.